rogue/src/main.rs

```diff
@@ -71,45 +70,117 @@ use general_movement_system::*;
 mod gamelog;
 use gamelog::GameLog;
 
+//--------------------------------------------------------------------
 // Debug flags.
-const DEBUG_DRAW_ALL_MAP: bool = false;
-const DEBUG_RENDER_ALL: bool = false;
+//--------------------------------------------------------------------
+const DEBUG_DRAW_ALL_MAP: bool = true;
+const DEBUG_RENDER_ALL: bool = true;
 const DEBUG_VISUALIZE_MAPGEN: bool = false;
 const DEBUG_HIGHLIGHT_STAIRS: bool = false;
 const DEBUG_HIGHLIGHT_FLOOR: bool = false;
 const DEBUG_HIGHLIGHT_FIRE: bool = false;
+const DEBUG_HIGHLIGHT_GRASS: bool = false;
 
 const MAPGEN_FRAME_TIME: f32 = 100.0;
 
+//--------------------------------------------------------------------
```

Comment:
We've swapped the main turn order FROM:

Player => Hazards => Monsters

TO:

Hazards => Player => Monster

The reasoning: terrain should have an immediate effect on any entities that spawn within it. I.e., if a snake spawned in tall grass, it should be *immediately* cloaked. There are still some subtle issues of turn order in the state machine lying in wait for us in the future, I'm sure. For now this seems to work well.